src/mds/Server.cc

```diff
@@ -4609,11 +4609,6 @@ void Server::handle_client_openc(MDRequestRef& mdr)
     // it existed.
     ceph_assert(mdr.get()->is_rdlocked(&dn->lock));
 
-    MutationImpl::LockOpVec lov;
-    lov.add_rdlock(&dnl->get_inode()->snaplock);
-    if (!mds->locker->acquire_locks(mdr, lov))
```

Comment:
It seems MDS allows us to acquire the same lock more than once. And even the `rdlock` has been already acquired it will succeeded here. So this shouldn't be a issue.